File: dv/screen/static/js/right_2.js
$.ajax({
    url: '/screen/manager_data',
    type: 'GET',
    dataType: 'json',
    contentType: 'application/json;charset=UTF-8',
    data: {},
    success: function (data) {
        console.log(data.class_name);
        console.log(data.class_sales);
        console.log(data.class_profit);
        // 基于准备好的dom，初始化echarts实例
        var myChart = echarts.init(document.getElementById('data_manager'));
        // 指定图表的配置项和数据
        var option = {
				tooltip: {
					trigger: 'axis',
					axisPointer: {
						type: 'shadow'
					}
				},
				legend: {
					data: ['vip新作月票', '总月票'],
					textStyle: {
                    color: "rgba(255, 250, 250, 1)"
                }
				},
				toolbox: {
					show: true,
					orient: 'vertical',
					x: 'right',
					y: 'center',
					feature: {
						mark: {
							show: true
						},
						dataView: {
							show: true,
							readOnly: false
						},
						magicType: {
							show: true,
							type: ['line', 'bar', 'stack', 'tiled']
						},
						restore: {
							show: true
						},
						saveAsImage: {
							show: true
						}
					}
				},
				xAxis: [{
					type: 'category',
					data: data.class_name
				}],
				yAxis: [{
					type: 'value'
				}],
				series: [

					{
						name: '总月票',
						type: 'bar',
						stack: 'manager',
						data: data.class_profit
					},
					{
						name: 'vip新作月票',
						type: 'bar',
						stack: 'manager',
						data: data.class_sales
					},
				],

			};
        myChart.setOption(option, true);
    }
});

File: pic/right2.py
import pandas as pd
import pymysql
from pyecharts import options as opts
from pyecharts.charts import Bar

conn = pymysql.connect(host='localhost', user='root', password='123456', database='qidian', charset='utf8')
cursor = conn.cursor()
sql = "SELECT * FROM `月票榜`"
cursor.execute(sql)
result = cursor.fetchall()
columns = [desc[0] for desc in cursor.description]
df = pd.DataFrame(result, columns=columns)
sql = "SELECT * FROM `vip新作月票榜`"
cursor.execute(sql)
result = cursor.fetchall()
columns = [desc[0] for desc in cursor.description]
new_df = pd.DataFrame(result, columns=columns)
cursor.close()
conn.close()
# 按照书籍类型分组并计算每种类型的数量
df = df.groupby('type').size()
df = df.reset_index().rename(columns={0: 'count'})
new_df = new_df.groupby('type').size()
new_df = new_df.reset_index().rename(columns={0: 'count'})
result = pd.merge(df, new_df, on='type')
manager = result['type'].tolist()
profit = result['count_x'].tolist()
sales = result['count_y'].tolist()
# 这里用一个示例数据来模拟你的数据
data = {
    "class_name": manager,
    "class_sales": profit,
    "class_profit": sales
}

# 初始化柱状图
bar = (
    Bar()
    .add_xaxis(data["class_name"])
    .add_yaxis("总月票", data["class_profit"], stack="manager")
    .add_yaxis("vip新作月票", data["class_sales"], stack="manager")
    .set_series_opts(label_opts=opts.LabelOpts(is_show=False))
    .set_global_opts(
        tooltip_opts=opts.TooltipOpts(trigger="axis", axis_pointer_type="shadow"),
        legend_opts=opts.LegendOpts(
            textstyle_opts=opts.TextStyleOpts(color="black")  # 设置图例文字颜色为黑色
        ),
        toolbox_opts=opts.ToolboxOpts(
            is_show=True,
            orient="vertical",
            pos_right="5%",
            pos_top="center",
            feature={
                "mark": {"show": True},
                "dataView": {"show": True, "readOnly": False},
                "magicType": {"show": True, "type": ["line", "bar", "stack", "tiled"]},
                "restore": {"show": True},
                "saveAsImage": {"show": True},
            },
        ),
        xaxis_opts=opts.AxisOpts(type_="category"),
        yaxis_opts=opts.AxisOpts(type_="value"),
    )
)

# 生成 HTML 文件
bar.render("right2.html")
